operator/pkg/manifest/renderer_cache_test.go

```diff
@@ -0,0 +1,230 @@
+	value, ok := m.config.Load(key)
+	if !ok {
+		return 0
+	}
+	return value.(uint32)
+}
+
+func (m *mockCache) SetConfig(key client.ObjectKey, cfg uint32) {
+	m.config.Store(key, cfg)
+	setConfigCount++
+}
+
+func (m *mockCache) DeleteConfig(key client.ObjectKey) {
+	m.config.Delete(key)
+}
+
+var _ = Describe("given manifest with a helm repo", Ordered, func() {
+	BeforeAll(func() {
+		rendererCache = &mockCache{
+			processor: sync.Map{},
+			config:    sync.Map{},
+		}
+	})
+	BeforeEach(func() {
+		// delete processor entry for parent key
+		rendererCache.DeleteProcessor(client.ObjectKey{Name: parentCacheKey, Namespace: testNs})
+		rendererCache.DeleteProcessor(client.ObjectKey{Name: parentCacheKey2, Namespace: testNs})
+		// delete processor entry for child key
+		rendererCache.DeleteConfig(client.ObjectKey{Name: testResourceName, Namespace: testNs})
+		rendererCache.DeleteConfig(client.ObjectKey{Name: testResourceName2, Namespace: testNs})
+		rendererCache.DeleteConfig(client.ObjectKey{Name: testResourceName3, Namespace: testNs})
+		// reset set counters
+		setConfigCount = 0
+		setProcessorCount = 0
+		// clear cached manifests
+		os.RemoveAll(util.GetFsManifestChartPath("../test_samples/kustomize"))
+		os.RemoveAll(util.GetFsManifestChartPath("../test_samples/helm"))
+	})
+
+	DescribeTable("given renderer cache for manifest processing",
+		func(testCaseFn func(resourceName string, parentKey string, flags types.ChartFlags, cache types.RendererCache,
+		) (string, string, uint32)) {
+			// first call for operations for same parent resource and configuration
+			verifyCacheEntries(testCaseFn(testResourceName, parentCacheKey, chartFlagsVariantOne, rendererCache))
+			// second call for operations for same parent resource and configuration
+			verifyCacheEntries(testCaseFn(testResourceName, parentCacheKey, chartFlagsVariantOne, rendererCache))
+			// third call for operations for same parent resource and configuration
+			verifyCacheEntries(testCaseFn(testResourceName, parentCacheKey, chartFlagsVariantOne, rendererCache))
+			// fourth call for operations for same parent resource and configuration
+			verifyCacheEntries(testCaseFn(testResourceName, parentCacheKey, chartFlagsVariantOne, rendererCache))
+			// check set count
+			Expect(setProcessorCount).To(Equal(2)) // new processor + new flags
+			Expect(setConfigCount).To(Equal(1))    // new flags
+
+			// fifth call for operations for same parent resource and DIFFERENT configuration
+			verifyCacheEntries(testCaseFn(testResourceName, parentCacheKey, chartFlagsVariantTwo, rendererCache))
+			// check set count
+			Expect(setProcessorCount).To(Equal(3)) // ^above + updated flags
+			Expect(setConfigCount).To(Equal(2))    // ^above + updated flags
+
+			// sixth call for operations for same parent DIFFERENT resource and configuration
+			verifyCacheEntries(testCaseFn(testResourceName2, parentCacheKey, chartFlagsVariantOne, rendererCache))
```

Comment:
It would be good if you add a chart flags variant (`chartFlagsVariantThree` for example) that includes `SetFlags` that you update them in the next call to `verifyCacheEntries()`

Reply:
I think that's be over-engineering, since the check is done on diff of `types.ChartFlags`, which contains both, `ConfigFlags` and `SetFlags`.

Reply:
I agree that the check is done on diff of `types.ChartFlags` values. But I would like to see them proved in the tests. Because the current `chartFlagsVariantOne` and `chartFlagsVariantTwo` do not include `SetFlags` values.

Reply:
As discussed. Thank you for adding some explanatory comments 👍